### PR TITLE
jpalmer108/updateEventApi/#156

### PR DESCRIPTION
**Pull request by jpalmer108**
# Alaska Airlines Pull Request

After discussing in-issue thread, I'm satisfied with adding these doc notes and wrapping up this issue. 

**Fixes:** #156 

## Summary:

More clear documentation regarding what can be gathered from the `input` event as well as the purpose for the `validated` event. This is so other's aren't also confused by it in the future. 

## Type of change:

Please delete options that are not relevant.
- [X] Other (Docs)

## Checklist:

- [X] My update follows the CONTRIBUTING guidelines of this project
- [X] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -41,8 +41,8 @@
 |-----------------------|--------------------------------------|--------------------------------------------------|
 | `auroInput-helpText`  | `CustomEvent<{ message: string; }>`  | Notifies that the helpText value has changed.    |
 | `auroInput-ready`     | `CustomEvent<any>`                   | Notifies that the component has finished initializing. |
-| `auroInput-validated` | `CustomEvent<{ isValid: boolean; }>` | Notifies that the isValid value has changed.     |
-| `input`               |                                      | Event fires when the value of an `auro-input` has been changed. |
+| `auroInput-validated` | `CustomEvent<{ isValid: boolean; }>` | Notifies that the isValid value has changed. Typically used for cross component communication of error states in more complex components. |
+| `input`               |                                      | Event fires when the value of an `auro-input` has been changed. Both updated string and `isValid` boolean are accessible. |
```
Comment (jason-capsule42) [inline docs/api.md:45]: All public attributes are accessible as a basic rule. `isValid` is always available to be checked regardless of if `input` event being fired. This seems redundant to me and muddies the explanation of this event.

File: src/base-input.js [hunk 2]
```diff
@@ -53,10 +53,10 @@ import i18n, {notifyOnLangChange, stopNotifyingOnLangChange} from './i18n.js';
  * @csspart helpText - Use for customizing the style of the helpText element
  * @csspart accentIcon - Use for customizing the style of the accentIcon element (e.g. credit card icon, calendar icon)
  * @csspart iconContainer - Use for customizing the style of the iconContainer (e.g. X icon for clearing input value)
- * @event input - Event fires when the value of an `auro-input` has been changed.
+ * @event input - Event fires when the value of an `auro-input` has been changed. Both updated string and `isValid` boolean are accessible.
```
Comment (jason-capsule42) [inline src/base-input.js:56]: All public attributes are accessible as a basic rule. `isValid` is always available to be checked regardless of if `input` event being fired. This seems redundant to me and muddies the explanation of this event.

File: src/base-input.js [hunk 3]
```diff
@@ -53,10 +53,10 @@ import i18n, {notifyOnLangChange, stopNotifyingOnLangChange} from './i18n.js';
...
  * @fires auroInput-helpText - Notifies that the helpText value has changed.
  * @fires auroInput-ready - Notifies that the component has finished initializing.
- * @fires auroInput-validated - Notifies that the isValid value has changed.
+ * @fires auroInput-validated - Notifies that the isValid value has changed. Typically used for cross component communication of error states in more complex components.
```
Comment (jason-capsule42) [inline src/base-input.js:59]: The event description should only contain a short statement about what the event is. Extended implementation details such as this should be in the demo documentation.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: src/base-input.js (hunk 3 above)
Comment (blackfalcon) [reply src/base-input.js:59]: See previous comment, same reply.

**Comment by blackfalcon** [comment]
@jpalmer108 please be sure to read the contribution guidelines for how to [construct a commit message](https://github.com/AlaskaAirlines/auro-input/blob/main/.github/CONTRIBUTING.md#git-commit-messages).

File: src/base-input.js (hunk 2 above)
Comment (jpalmer108) [reply src/base-input.js:56]: Got it! I can check that out.

**Comment by jpalmer108** [comment]
> apiExamples

I'm not sure I have the context to understand what you're referencing with adding the `apiExamples.md` doc, however I can add this update to the `demo.md` and remove my updates from this file. Will someone else with context on the `apiExamples.md`doc be able to implement your suggestion here?

**Comment by jason-capsule42** [comment]
@jpalmer108 This work needs to happen in the apiExamples.md file (which needs to be added). It's not obvious how to do it because this repo is a little out of date with our latest generator. 

If you would like, I can make some time with you today and we can update this repo together. It will be a little bit of work though. Otherwise, I can take this change on. It's all documentation, so it shouldn't impact your ability to ship, correct?

**Comment by jpalmer108** [comment]
> It's all documentation, so it shouldn't impact your ability to ship, correct?

Correct, we are unblocked.

**Pull request closed without merging**